### PR TITLE
Remove 9.6

### PR DESCRIPTION
**Pull request by jasonmp85**
DESCRIPTION: Removes support and code for PostgreSQL 9.6

### CODE REVIEW

**Comment by mtuncer** [comment]
Can we continue with this work ? Travis is running extra PG 9.6 for each commit. Could we at least remove 9.6 from build matrix and merge that change without waiting for the rest of the changes ?

**Comment by marcocitus** [comment]
> Can we continue with this work ? Travis is running extra PG 9.6 for each commit. Could we at least remove 9.6 from build matrix and merge that change without waiting for the rest of the changes ?

I'd be ok with pushing https://github.com/citusdata/citus/pull/2554/commits/34e495a8e090afa4a3408bbb439753de576b6d59 straight to master.

**Comment by jasonmp85** [comment]
> Can we continue with this work?

Yes. I'm back from winter break and plan on addressing all feedback today.

**Comment by metdos** [comment]
> Yes. I'm back from winter break and plan on addressing all feedback today.

@jasonmp85 any update on this?

File: src/backend/distributed/commands/create_distributed_table.c [hunk 1]
```diff
@@ -1191,11 +1191,7 @@ RegularTable(Oid relationId)
 {
```
[the remark this answers is not in the document]
Comment (jasonmp85) [reply src/backend/distributed/commands/create_distributed_table.c]: Updated it.

File: src/backend/distributed/commands/create_distributed_table.c [hunk 2]
```diff
@@ -1388,7 +1384,6 @@ TupleDescColumnNameList(TupleDesc tupleDescriptor)
 static bool
```
[the remark this answers is not in the document]
Comment (jasonmp85) [reply src/backend/distributed/commands/create_distributed_table.c]: Keeping this one, but changed _supported_ to _introduced in_.

File: src/backend/distributed/commands/schema.c [hunk 3]
```diff
@@ -147,13 +147,8 @@ GetSchemaNameFromDropObject(ListCell *dropSchemaCell)
 {
```
[the remark this answers is not in the document]
Comment (jasonmp85) [reply src/backend/distributed/commands/schema.c]: Good catch. I've now done precisely that (removed it).

File: src/backend/distributed/commands/table.c [hunk 4]
```diff
@@ -129,7 +129,6 @@ ProcessDropTableStmt(DropStmt *dropTableStatement)
 void
```
[the remark this answers is not in the document]
Comment (jasonmp85) [reply src/backend/distributed/commands/table.c]: Fixed.

File: src/backend/distributed/commands/table.c [hunk 5]
```diff
@@ -194,7 +192,6 @@ ProcessCreateTableStmtPartitionOf(CreateStmt *createStatement)
...
```
[the remark this answers is not in the document]
Comment (jasonmp85) [reply src/backend/distributed/commands/table.c]: (In the future, leave GitHub comments _on the line in question_. Putting them on the `void` line like this means I don't see the comment at all in GitHub without expanding some block or other in the UI).

Fixed.

File: src/backend/distributed/commands/transmit.c [hunk 6]
```diff
@@ -107,25 +103,16 @@ SendRegularFile(const char *filename)
```
[the remark this answers is not in the document]
Comment (jasonmp85) [reply src/backend/distributed/commands/transmit.c]: If something mentions Hadoop, it is _quite_ stale indeed.

Marking as resolved as there's no action for me here.

File: src/backend/distributed/commands/utility_hook.c [hunk 7]
```diff
@@ -96,16 +74,12 @@ void
 CitusProcessUtility(Node *node, const char *queryString, ProcessUtilityContext context,
```
[the remark this answers is not in the document]
Comment (jasonmp85) [reply src/backend/distributed/commands/utility_hook.c]: Changed the comment to describe it as a convenience method to wrap arguments in a `PlannedStmt` before passing along `ProcessUtility`.

File: src/backend/distributed/connection/remote_commands.c [hunk 8]
```diff
@@ -824,13 +800,8 @@ WaitForAllConnections(List *connectionList, bool raiseInterrupts)
 			}
```
[the remark this answers is not in the document]
Comment (jasonmp85) [reply src/backend/distributed/connection/remote_commands.c]: PLEASE comment _on the line you're talking about_. This is getting very confusing.

File: src/backend/distributed/transaction/citus_dist_stat_activity.c [hunk 9]
```diff
@@ -125,149 +123,75 @@
  * Also, backend_type introduced with pg 10+ so we have null in the previous verions.
  */
 
-#if PG_VERSION_NUM >= 100000
```
[the remark this answers is not in the document]
Comment (jasonmp85) [reply src/backend/distributed/transaction/citus_dist_stat_activity.c]: Fixed.

**Comment by jasonmp85** [comment]
> @jasonmp85 any update on this?

@metdos I just pushed a commit addressing all feedback; sorry, was a bit delayed due to home office move, screwed up laptop workspace (long story), and CircleCI work.

Since all of the feedback was on comments, I'm guessing the code was fine… going to merge this once it passes. @velioglu: if you _did_ have more you wanted feedback on, let me know. Since this is such a big patch, I don't want it sitting around any longer, though.

**Comment by jasonmp85** [comment]
Removes support and code for PostgreSQL 9.6

**Pull request merged**